diff -Nur wapd-0.8/Makefile wapd-0.8-n/Makefile
--- wapd-0.8/Makefile	2006-11-07 02:11:51.000000000 +0100
+++ wapd-0.8-n/Makefile	2006-11-07 03:04:48.000000000 +0100
@@ -16,7 +16,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual -ggdb
+CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual -ggdb
 
 ### The directory environment:
 
diff -Nur wapd-0.8/server.c wapd-0.8-n/server.c
--- wapd-0.8/server.c	2005-12-24 15:19:20.000000000 +0100
+++ wapd-0.8-n/server.c	2006-11-07 03:04:48.000000000 +0100
@@ -694,7 +694,7 @@
            int offset = strtol(getopt, NULL, 10);
            int i = 0;
            const cEvent *PresentEvent =
-              Schedule->GetPresentEvent(channel->Number() == cDevice::CurrentChannel());
+              Schedule->GetPresentEvent();
            time_t now = time(NULL) - Setup.EPGLinger * 60;
            for (const cEvent *Event = Schedule->Events()->First();
                 Event;
